common/windows/win32_window.h

```diff
@@ -0,0 +1,92 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef WIN32_WINDOW_H_
+#define WIN32_WINDOW_H_
+
+#include <Windows.h>
+#include <Windowsx.h>
+
+#include <memory>
+#include <string>
+
+// A class abstraction for a high DPI aware Win32 Window.  Intended to be
+// inherited from by classes that wish to specialize with custom
+// rendering and input handling
+class Win32Window {
+ public:
+  Win32Window();
+  ~Win32Window();
+
+  // Creates and shows a win32 window with |title| and position and size using |x|,
+  // |y|, |width| and |height|
+  bool CreateAndShow(const char *title, const unsigned int x, const unsigned int y,
+                  const unsigned int width, const unsigned int height);
+
+  // Release OS resources asociated with window.
```

Comment:
Can you elaborate on what makes this conceptually different from the destructor?